pkg/collector/component.go

```diff
@@ -0,0 +1,277 @@
+package collector
+
+import (
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/cors"
+	"github.com/spf13/viper"
+	"github.com/uber/jaeger-lib/metrics"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/thrift"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/builder"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/grpcserver"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/zipkin"
+	clientcfgHandler "github.com/jaegertracing/jaeger/pkg/clientcfg/clientcfghttp"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+	"github.com/jaegertracing/jaeger/pkg/recoveryhandler"
+	ss "github.com/jaegertracing/jaeger/plugin/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/plugin/storage"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	jc "github.com/jaegertracing/jaeger/thrift-gen/jaeger"
+	sc "github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	zc "github.com/jaegertracing/jaeger/thrift-gen/zipkincore"
+)
+
+var _ (io.Closer) = (*Component)(nil)
+
+// Component returns the collector as a manageable unit of work
+type Component struct {
```

Comment:
I would call this `collector.Collector`. Calling it `collector.Component` makes it unnecessarily vague.

Reply:
What about the package? I think `pkg/collector` is really wrong, as this depends on the `cmd/collector/builder` and `cmd/collector/app` packages, but I don't have a good idea for a package name for this...

Reply:
We can put it directly under cmd/collector/app, similar to Agent type.

Reply:
Alright, will give it a try.